### PR TITLE
Translation: German

### PR DESCRIPTION
**Pull request by koko-joe**
Translated to german, open for discussion

### CODE REVIEW

**Comment by jwasham** [comment]
I'll leave this open until you're ready for pull.

**Comment by koko-joe** [comment]
I translated everything except book titles or video titles and ran German spell checker over it.
I could check if there are German translations for these books if necessary.
I also could check on some links to Wikipedia if there are German translations.
The explanations on German Wikipedia pages can differ quite a lot from the English versions.
Many other translations did not do that kind of work.
If there is no German speaking reviewer, then I would like you to merge this pull request.

File: translations/README-de.md [hunk 1]
```diff
@@ -0,0 +1,2084 @@
...
+
+</details>
+
+## Worum es geht
+
+Das ist mein mehrmonatiger Lernplan um vom Web Developer (Selbststudium, kein Abschluss in Informatik) zum Software Engineer bei einer großen Firma zu wechseln.
+
+![Coding at the whiteboard - from HBO's Silicon Valley](https://d3j2pkmjtin6ou.cloudfront.net/coding-at-the-whiteboard-silicon-valley.png)
+
+Dies ist gedacht für **neue Software Engineure** oder solche die von der Software/Web Entwicklung zum Software
```
Comment (cndolo) [inline translations/README-de.md]: This should read "Softwareingenieur".

File: translations/README-de.md [hunk 2]
```diff
@@ -0,0 +1,2084 @@
...
+
+Das ist mein mehrmonatiger Lernplan um vom Web Developer (Selbststudium, kein Abschluss in Informatik) zum Software Engineer bei einer großen Firma zu wechseln.
+
+![Coding at the whiteboard - from HBO's Silicon Valley](https://d3j2pkmjtin6ou.cloudfront.net/coding-at-the-whiteboard-silicon-valley.png)
+
+Dies ist gedacht für **neue Software Engineure** oder solche die von der Software/Web Entwicklung zum Software 
+Engineering wechseln wollen (wobei Informatikkenntnisse benötigt werden). Falls du behauptest mehrere Jahre an Erfahrung
+als Software Engineer zu haben, erwartet dich ein hartes Vorstellungsgespräch.
+
+Falls du schon mehrere Jahre Erfahung in der Software/Webenteicklung hast, muss dir klar sein, dass große Software
```
Comment (cndolo) [inline translations/README-de.md]: Typo in "Webentwicklung".

File: translations/README-de.md [hunk 3]
```diff
@@ -0,0 +1,2084 @@
...
+## Inhaltsverzeichnis
+
+- [Worum es geht](#what-is-it)
+- [Warum solltest du das hier lesen?](#why-use-it)
+- [Wie man dies hier benutzt](#how-to-use-it)
+- [Glaub nicht, dass du nicht klug genug bist](#dont-feel-you-arent-smart-enough)
+- [Über Video Resourcen](#about-video-resources)
+- [Ablauf von Vorstellungsgesprächen und allgemeine Vorbereitung darauf](#interview-process--general-interview-prep)
+- [Wähle eine Sprache für das Vorstellungsgespräch](#pick-one-language-for-the-interview)
+- [Buchliste](#book-list)
```
Comment (cndolo) [inline translations/README-de.md]: I'd suggest to change this to "Literaturliste".

File: translations/README-de.md [hunk 4]
```diff
@@ -0,0 +1,2084 @@
...
+- [Graphen](#graphs)
+    - gerichtet
+    - ungerichtet
+    - Adjazenzmatrix
+    - Adjazenzliste
+    - Traversierung: Breitensuche, Tiefensuche
+- [Sonstiges](#even-more-knowledge)
+    - [Rekursion](#recursion)
+    - [Dynamische Programmierung](#dynamic-programming)
+    - [Object-orientierte Programmierung](#object-oriented-programming)
```
Comment (cndolo) [inline translations/README-de.md]: This should be "Objektorientierte Programmierung".

File: translations/README-de.md [hunk 5]
```diff
@@ -0,0 +1,2084 @@
...
+    - [Design Patterns (Entwurfsmuster)](#design-patterns)
+    - [Kombinatorik (n über k) und Wahrscheinlichkeiten](#combinatorics-n-choose-k--probability)
+    - [NP, NP-Vollständig und Heuristiken](#np-np-complete-and-approximation-algorithms)
+    - [Caches](#caches)
+    - [Proczsse und Threads](#processes-and-threads)
```
Comment (cndolo) [inline translations/README-de.md]: Type in "Prozesse"

File: translations/README-de.md [hunk 6]
```diff
@@ -0,0 +1,2084 @@
...
+- [Wenn du den Job bekommst](#once-youve-got-the-job)
+
+---------------- Alles unter der Linie ist optional ----------------
+
+<details>
+<summary>Additional Resources</summary>
+
+- [Zusätzliche Bücher](#additional-books)
+- [Zusätzliches Wissen](#additional-learning)
+    - [Compilers (Übersetzer)](#compilers)
```
Comment (cndolo) [inline translations/README-de.md]: Compiler.

File: translations/README-de.md [hunk 7]
```diff
@@ -0,0 +1,2084 @@
...
+    - [Mathematik für schnelle Berechnungen](#math-for-fast-processing)
+    - [Treap](#treap)
+    - [Lineare Programmierung](#linear-programming-videos)
+    - [Geometrie, Konvexe Hülle](#geometry-convex-hull-videos)
+    - [Diskrete Mathematik](#discrete-math)
+    - [Machine Learning (maschinelles Lernen)](#machine-learning)
+- [Weitere Details für ausgewählte Themen](#additional-detail-on-some-subjects)
+- [Videoreihen](#video-series)
+- [Infomatikkurse](#computer-science-courses)
+- [Wissenschaftliche Abhandlungen](#papers)
```
Comment (cndolo) [inline translations/README-de.md]: Or maybe "Wissenschaftliche Veröffentlichtungen."

**Comment by koko-joe** [comment]
Thanks @cndolo! 
Fixed the typos.

**Comment by koko-joe** [comment]
Hey @jwasham , since this PR was reviewed by someone. Would you be so kind and merge it?

**Comment by jwasham** [comment]
Thanks!

**Pull request merged**